ipatests/test_integration/test_otp.py

```diff
@@ -83,13 +85,71 @@ def kinit_otp(host, user, *, password, otp, success=True):
     )
 
 
+def ssh_2fa_with_cmd(host, username, password, otpvalue, command="exit 0"):
+    """ ssh to user and in same session pass the command to check tgt of user
+    :param host: host to ssh
+    :param str username: The name of user
+    :param str password: password, usually the first factor
+    :param str otpvalue: generated pin of user
+    :param str command: command to execute in same session,
+     by deafult set to "exit 0"
+    :return: object class of expect command run
+    """
+    temp_conf = tempfile.NamedTemporaryFile(suffix='.exp', delete=False)
+    with open(temp_conf.name, 'w') as tfile:
+        tfile.write('proc exitmsg { msg code } {\n')
+        tfile.write('\t# Close spawned program, if we are in the prompt\n')
+        tfile.write('\tcatch close\n\n')
+        tfile.write('\t# Wait for the exit code\n')
+        tfile.write('\tlassign [wait] pid spawnid os_error_flag rc\n\n')
+        tfile.write('\tputs ""\n')
+        tfile.write('\tputs "expect result: $msg"\n')
+        tfile.write('\tputs "expect exit code: $code"\n')
+        tfile.write('\tputs "expect spawn exit code: $rc"\n')
+        tfile.write('\texit $code\n')
+        tfile.write('}\n')
+        tfile.write('set timeout 60\n')
+        tfile.write('set prompt ".*\\[#\\$>\\] $"\n')
+        tfile.write(f'set password "{password}"\n')
+        tfile.write(f'set otpvalue "{otpvalue}"\n')
+        tfile.write(f'spawn ssh -o NumberOfPasswordPrompts=1 -o '
+                    f'StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'
+                    f' -l {username} localhost {command}\n')
```

Comment:
In [/var/log/sssd/sssd_ssh.log](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/3e215fe4-5ba1-11ef-8ef5-fa163ed57a39/test_integration-test_otp.py-TestOTPToken-test_2fa_only_with_password/master.ipa.test/var/log/sssd/sssd_ssh.log.gz) we see the following error:
```
(2024-08-16  7:40:01): [ssh] [cache_req_search_dp] (0x0400): [CID#1] CR #0: Looking up [localhost] in data provider
(2024-08-16  7:40:01): [ssh] [cache_req_common_process_dp_reply] (0x3f7c0): [CID#1] CR #0: Data Provider Error: 3, 22, Invalid argument
(2024-08-16  7:40:01): [ssh] [cache_req_common_process_dp_reply] (0x0400): [CID#1] CR #0: Due to an error we will return cached data
(2024-08-16  7:40:01): [ssh] [sss_domain_get_state] (0x1000): [CID#1] Domain ipa.test is Active
(2024-08-16  7:40:01): [ssh] [cache_req_search_cache] (0x0400): [CID#1] CR #0: Looking up [localhost] in cache
(2024-08-16  7:40:01): [ssh] [sysdb_search_ssh_hosts] (0x0400): [CID#1] No such host
(2024-08-16  7:40:01): [ssh] [cache_req_search_cache] (0x0400): [CID#1] CR #0: Object [localhost] was not found in cache
(2024-08-16  7:40:01): [ssh] [cache_req_process_result] (0x0400): [CID#1] CR #0: Finished: Not found
(2024-08-16  7:40:01): [ssh] [sysdb_search_ssh_hosts] (0x0400): [CID#1] No such host
```
Maybe you can try to replace localhost with the actual host name (host.hostname).

Reply:
Okay, I did not check the log, but authentication worked while testing the code.